### ISSUE TITLE
MATCH Query : Access array offset on value of type null in DQL parser

### ISSUE THREAD

**Issue by aguidis** [doctrine/orm#8222]
## Bug Report

| Q       | A      |
|---------|--------|
| BC      | no     |
| Version | v2.7.3 |
| PHP | 7.4.8 |

## Summary

I installed the [DoctrineExtensions](https://github.com/beberlei/DoctrineExtensions) to use the [**MATCH AGAINST**](https://github.com/beberlei/DoctrineExtensions/blob/master/src/Query/Mysql/MatchAgainst.php) query.

I added this clause in my repository method:

```php
if ($name = $search->name) {
    $qb
        ->andWhere('MATCH(shop.name) AGAINST (:name BOOLEAN)')
        ->setParameter('name', $name.'*')
    ;
}
```

## Current behavior

> Notice: Trying to access array offset on value of type null

![image](https://user-images.githubusercontent.com/1856652/89124139-e6e2cd00-d4d4-11ea-8077-e6e3aa934b54.png)

When the `MATCH` clause is parsed, the following error is triggered in the [Parser](https://github.com/doctrine/orm/blob/2.7/lib/Doctrine/ORM/Query/Parser.php#L3323) because `$this->lexer->lookahead` is `null`.

![image](https://user-images.githubusercontent.com/1856652/89124014-f9103b80-d4d3-11ea-9290-a89a98236279.png)

**Comment by beberlei** [comment]
@greg0ire this sounds a lot like that bug we had early in 2.7 cycle, do you remember what it was?

**Comment by greg0ire** [comment]
@beberlei it doesn't ring a bell but I found this in the issues: https://github.com/doctrine/orm/issues/7983

**Comment by bigwhoop** [comment]
Hi. I had this error in v2.7.3 as well, but it seems to be fixed with v2.7.4.

**Comment by greg0ire** [comment]
@aguidis can you try upgrading?

**Comment by Bruspashko** [comment]
@greg0ire I'm having the same problem on 2.7.4. It's basically because Parser doesn't know how to handle Match Against. It expects a comparison operator, and nothing is passed.

**Comment by bigwhoop** [comment]
IIRC correctly the query needs to have a comparison like `= 1` in boolean mode. Example:

```
->andWhere('MATCH(x.field) AGAINST (:value BOOLEAN) = 1')
->setParameter('value', $value)
```

**Comment by julkwel** [comment]
@bigwhoop  this fix the error but return no results ( i don't know if i have a mistake in this query ).

I'm sure that I have this text in title column ` Lorem ipsum need to query ` , and my query 
```
$qb->createQueryBuilder('c')
->andWhere("MATCH (c.title) AGAINST(:value BOOLEAN) = 1")
->setParameter('value', 'ipsum')->getQuery()->getResults(); 
``` 

return empty.

**Comment by pgrzesiecki** [comment]
Yes it return no data because comparing this to `= 1` or `= 0` this query is not valid `match against` query (especially in mariadb).

**Comment by mklewitz** [comment]
MATCH AGAINST doesn't simply return 0 or 1, but a score. So this will work:

```MATCH (alias.fieldName) AGAINST(:searchValue BOOLEAN) > 0```

You can even select the score as scalar value and/or order by the best match:

```
$this->createQueryBuilder('alias')
    ->addSelect('MATCH(alias.fieldName) AGAINST (:searchValue BOOLEAN)')
    ->where('MATCH(alias.fieldName) AGAINST (:searchValue BOOLEAN) > 0')
    ->orderBy('MATCH(alias.fieldName) AGAINST (:searchValue BOOLEAN)', 'DESC')
    ->setParameter('searchValue', $searchValue)
```
